tests/migration_test.go

```diff
@@ -931,24 +932,59 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		})
 		Context("with an Fedora shared NFS PVC, cloud init and service account", func() {
 			var pvName string
+			var nfsPodName string
 			var vmi *v1.VirtualMachineInstance
-			BeforeEach(func() {
+
+			initNFS := func(ipFamily k8sv1.IPFamily) {
 				tests.SkipNFSTestIfRunnigOnKindInfra()
 				pvName = "test-nfs" + rand.String(48)
 				// Prepare a NFS backed PV
 				By("Starting an NFS POD")
 				os := string(cd.ContainerDiskFedora)
-				nfsIP := tests.CreateNFSTargetPOD(os)
+
 				// create a new PV and PVC (PVs can't be reused)
 				By("create a new NFS PV and PVC")
+				nfsPod := tests.CreateNFSTargetPOD(os)
+				nfsPodName = nfsPod.Name
+				nfsIP := libnet.GetPodIpByFamily(nfsPod, ipFamily)
+				ExpectWithOffset(1, nfsIP).NotTo(BeEmpty())
 				tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
+			}
+
+			BeforeEach(func() {
+				pvName = ""
+				nfsPodName = ""
```

Comment:
This is again optional, but I would rather have pointers to these, so you could compare to nil.

Reply:
I changed the `AfterEach`es code to check whther `nfsInitialized`. So there is no need to compare to the names.